src/structures/GuildBan.js

```diff
@@ -0,0 +1,64 @@
+'use strict';
+
+const Base = require('./Base');
+
+/**
+ * Represents a ban in a guild on Discord.
+ * @extends {Base}
+ */
+class GuildBan extends Base {
+  /**
+   * @param {Client} client The instantiating client
+   * @param {Object} data The data for the ban
+   * @param {Guild} guild The guild in which the ban is
+   */
+  constructor(client, data, guild) {
+    super(client);
+
+    /**
+     * The guild in which the ban is
+     * @type {Guild}
+     */
+    this.guild = guild;
+
+    /**
+     * The reason for the ban
+     * @type {?string}
+     */
+    this.reason = null;
+
+    /**
+     * The user this ban applies to
+     * @type {User}
+     */
+    this.user = null;
+
+    if (data) this._patch(data);
+  }
+
+  _patch(data) {
+    if ('user' in data) this.user = this.client.users.add(data.user, true);
+    if ('reason' in data) this.reason = data.reason;
+  }
+
+  /**
+   * Whether this GuildBan is a partial
+   * If the reason is not provided the value is null
+   * @type {boolean}
+   * @readonly
+   */
+  get partial() {
+    return this.reason === undefined;
```

Comment:
```suggestion
    return this.reason === null;
```

Reply:
I don't see why this needs changing, if the reason is `null` then there is no reason, if its `undefined` then its missing (partial)

Reply:
i see how it can be null if no reason is provided, but it can't really be undefined, can it? (because you're setting it to null in the constructor)

Reply:
it can, GuildBanAdd calls `guild.bans.add({ user: data.user, reason: data.reason })` and then its set in the _patch method of GuildBan

Reply:
i am talking about https://github.com/discordjs/discord.js/blob/b7aad426be20fe89f2f2569449c03de636767216/src/structures/GuildBan.js#L28 which sets it to null even if not present in the provided data object

Reply:
yeah but it will always be in the data object (since its only called from GuildBanAdd (which always provides it)), I guess I could change the default value to undefined? 🤔

Reply:
the client events dont provide a reason, only guild_id and user

Reply:
the reason is not set to null in the constructor anymore so I'll mark this as resolved